query-node/schema.graphql

```diff
@@ -12,30 +12,142 @@ type Block @entity {
   network: Network!
 }
 
+enum MembershipEntryMethod {
+  PAID
+  SCREENING
+  GENESIS
+}
+
 "Stored information about a registered user"
-type Member @entity {
+type Membership @entity {
   "MemberId: runtime identifier for a user"
   id: ID!
 
   "The unique handle chosen by member"
-  handle: String @unique @fulltext(query: "membersByHandle")
+  handle: String! @unique @fulltext(query: "membersByHandle")
 
   "A Url to member's Avatar image"
   avatarUri: String
 
   "Short text chosen by member to share information about themselves"
   about: String
 
-  "Blocknumber when member was registered"
-  registeredAtBlock: Int!
-
   "Member's controller account id"
   controllerAccount: Bytes!
 
   "Member's root account id"
   rootAccount: Bytes!
 
-  happenedIn: Block!
+  "Blocknumber when member was registered"
+  registeredAtBlock: Block!
+
+  "Timestamp when member was registered"
+  registeredAtTime: BigInt!
+
+  "How the member was registered"
+  entry: MembershipEntryMethod!
+
+  "The type of subscription the member has purchased if any."
+  subscription: BigInt
+}
+
+# TODO: add description to fields
+# TODO: revise property requirement for the new and existing fields derived from content-metadata-protobuf
+
+type ChannelCategory @entity {
+  id: ID!
+
+  name: String
+}
+
+type Person @entity {
+  id: ID!
+
+  firstName: String
+
+  middleName: String
+
+  lastName: String
+
+  about: String
+
+  coverPhotoUrl: Asset
+
+  avatarPhotoUrl: Asset
+
+  videos: [Video!] @derivedFrom(field: "people")
+
+  seasons: [Season!] @derivedFrom(field: "people")
+
+  series: [Series!] @derivedFrom(field: "people")
+}
+
+type Playlist @entity {
+  id: ID!
+
+  title: String
+
+  videos: [Video!]
+}
+
+type Season @entity {
+  id: ID!
+
+  title: String
+
+  description: String
+
+  coverPhoto: String
+
+  people: [Person!]
+
+  episodes: [Video!]
+
+  series: Series @derivedFrom(field: "seasons")
+}
+
+type Series @entity {
+  id: ID!
+
+  in_channel: Channel!
+
+  title: String
+
+  description: String
+
+  coverPhoto: String
+
+  people: [Person!]
+
+  seasons: [Season!]
+}
+
+union Asset = AssetUrl | AssetStorage
```

Comment:
An Asset may also not be provided at all so it can be null

Reply:
Asset is nullable already. It can't be seen in the asset definition itself, but everywhere it is used, it is nullable (`myProperty: Asset` - without exclamation mark that marks non-nullable properties).